### ISSUE TITLE
Select accessibility issue for colorblind people

### ISSUE THREAD

**Issue by TheMineWay** [ant-design/ant-design#53749]
### Reproduction link

[https://ant.design/components/select#select-demo-variant](https://ant.design/components/select#select-demo-variant)

### Steps to reproduce

Use the Select component with the status prop. By default it only changes border color, which violates the W3C a11y guide https://www.w3.org/WAI/WCAG21/Understanding/use-of-color.html. The status color should be complemented with another element different than a color (an icon, for example).

### What is expected?

Color should not be the only status indicator. Colorblind people might not differentiate status colors, or even notice that teh status is different than the default.

### What is actually happening?

Only color is being applied to borders by default.

| Environment | Info |
| --- | --- |
| antd | 5.24.9 |
| React | Any |
| System | Any |
| Browser | Any |

---
By default, it could be nice to have an icon inside the Select component that shows status.

**Comment by MadCcc** [comment]
We may provide a 3A Theme in 6.x.

**Comment by TheMineWay** [comment]
> We may provide a 3A Theme in 6.x.

Shouldn't this be the default? I mean, I agree to move this to antd 6 as big changes might be coming, but I think the important part of this is to comply by default, then, enable customizations (theme config is a good place) so you can disable this. What do you think?
